internal/core/metadata/rest_devicereport.go

```diff
@@ -34,8 +33,7 @@ func restGetAllDeviceReports(w http.ResponseWriter, _ *http.Request) {
 	}
 
 	// Check max limit
-	if len(res) > Configuration.Service.MaxResultCount {
-		err = errors.New("Max limit exceeded")
+	if err = checkMaxLimit(Configuration.Service.MaxResultCount); err != nil {
```

Comment:
Should we be passing in `len(res)` here rather than `Configuration.Service.MaxResultCount` so that we verify the length of the response is not exceeding the max value allowed?